src/main/scala/com/codacy/client/bitbucket/client/BitbucketClientBase.scala

```diff
@@ -34,9 +34,48 @@ abstract class BitbucketClientBase(credentials: Credentials) {
     }
   }
 
-  /*
-   * Does a paginated API request and parses the json output into a sequence of classes
-   */
+  /**
+    * Make an API request and parse the json output into a response. This response includes not only the objects
+    * requested but pagination information as well.
+    *
+    * @param url The url to be invoked if no cursor is provided
+    * @param pageRequest The pagination request with cursor information
+    * @param reader A [[Reads]] to parse the json response from the API
+    * @tparam T The type of the objects in the response (excluding pagination information)
+    * @return A [[SuccessfulResponse]] or a [[FailedResponse]] depending if the request was successful or not
+    */
+  def executeWithCursor[T](url: String, pageRequest: PageRequest)(
+      implicit reader: Reads[T]
+  ): RequestResponse[Seq[T]] = {
+
+    val requestUrl = pageRequest.cursor match {
```

Comment:
Maybe a comment saying that when we have a cursor, it is actually already a fully built link to the next page